src/lib/operators/aggregate.cpp

```diff
@@ -605,14 +605,19 @@ write_aggregate_values(std::shared_ptr<ValueColumn<AggregateType>> column,
   auto& values = column->values();
   auto& null_values = column->null_values();
 
+  values.resize(results->size());
+  null_values.resize(results->size());
+
+  size_t i = 0;
   for (auto& kv : *results) {
-    null_values.push_back(!kv.second.current_aggregate);
+    null_values[i] = !kv.second.current_aggregate;
 
     if (!kv.second.current_aggregate) {
-      values.push_back(AggregateType());
+      values[i] = AggregateType();
```

Comment:
In case we do keep `resize()`, we can skip this line, since `values[i]` should already contain a default constructed `AggregateType()`.